packages/ember-htmlbars/lib/component.js

```diff
@@ -138,6 +139,21 @@ const Component = View.extend(TargetActionSupport, {
     }
   }),
 
+  templateForName(name, type) {
```

Comment:
@rwjblue yes, but there isn't any reason they couldn't have both used the same technique.  The same glimmer pattern could have been easily moved to where component manager in htmlbars does get('layout') that happens in ensureCompilable.